WeartGloves/src/WeArtGlove.cpp

```diff
@@ -0,0 +1,1598 @@
+    }
+    yInfo() << LogPrefix << "finger-tip frame names: " << fingerTipFrameNameList;
+
+    // Get human hand link name
+    if (!(config.check("hand_link") && config.find("hand_link").isString())) {
+        yError() << LogPrefix << "Unable to find hand_link in the config file.";
+        return false;
+    }
+    humanHandLinkName = config.find("hand_link").asString();
+    yInfo() << LogPrefix << "human hand link name: " << humanHandLinkName;
+
+    // Get human hand finger names
+    yarp::os::Bottle* fingerListYarp;
+    if (!(config.check("human_finger_list") && config.find("human_finger_list").isList()))
+    {
+        yError() << LogPrefix << "Unable to find human_finger_list in the config file.";
+        return false;
+    }
+    fingerListYarp = config.find("human_finger_list").asList();
+
+    for (size_t i = 0; i < fingerListYarp->size(); i++)
+    {
+        humanFingerNames.push_back(fingerListYarp->get(i).asString());
+    }
+    yInfo() << LogPrefix << "human finger names: " << humanFingerNames;
+
+    // Fill the coupling Matrix
+    yarp::os::Bottle* closureToHumanJointYarp;
+    if (trackingAlgorithmType == "SCTA")
+    {
+        if (!(config.check("closure_to_human_joint_simple")) || !config.find("closure_to_human_joint_simple").isList())
+        {
+            yError() << LogPrefix << "couldn't find the proper coupling matrix";
+            return false;
+        }
+        closureToHumanJointYarp = config.find("closure_to_human_joint_simple").asList();
+
+        if (closureToHumanJointYarp->size() < 60)
+        {
+            yError() << LogPrefix << "size of the coupling matrix for simple tracking is:" << closureToHumanJointYarp->size() << "Required 60";
+            return false;
+        }
+        closureToHumanJoint.resize(20, N_THIMBLES);
+        for (int j = 0; j < closureToHumanJointYarp->size(); j++)
+        {
+            int col = j % N_THIMBLES;
+            int row = j / N_THIMBLES;
+            closureToHumanJoint(row, col) = closureToHumanJointYarp->get(j).asFloat64();
+        }
+    }
+    else if (trackingAlgorithmType == "ASTA")
+    {
+        if (!(config.check("closure_to_human_joint_rawSensor")) || !config.find("closure_to_human_joint_rawSensor").isList())
+        {
+            yError() << LogPrefix << "couldn't find the proper coupling matrix";
+            return false;
+        }
+        closureToHumanJointYarp = config.find("closure_to_human_joint_rawSensor").asList();
+
+        if (closureToHumanJointYarp->size() < 260)
+        {
+            yError() << LogPrefix << "size of the coupling matrix for rawSensor tracking is:" << closureToHumanJointYarp->size() << "Required 260";
+            return false;
+        }
+        closureToHumanJoint.resize(20, 13);
+        for (size_t j = 0; j < closureToHumanJointYarp->size(); j++)
+        {
+            size_t col = j % consideredJointNameList.size();
+            size_t row = j / consideredJointNameList.size();
```

Comment:
Please define 20 and 13 as const variables. Then 260 is the multiplication of the two

Reply:
also, those numbers are derived from these matrices: 
`humanJointState.size()`, `consideredJointNameList.size()`. 
Do you think I can use these instead of defining const variables?

Reply:
Yes